source/os/source/threads.cpp

```diff
@@ -207,7 +197,7 @@ bool Signal::check(int checkFor) const {
 }
 
 bool Signal::checkAndSignal(int waitFor, int newSignal) {
-	return flag.compare_exchange_strong(newSignal, waitFor) == waitFor;
+	return flag.compare_exchange_strong(waitFor, newSignal);
```

Comment:
the std::atomic version of compare_exchange_strong has the arguments switched, and returns bool instead of the value currently being stored. 

The first argument is also a reference, and has it's value replaced with the value stored in the atomic object at the time the operation is initiated. Previously this would have been the return value.